src/vs/workbench/parts/preferences/common/keybindingsEditorModel.ts

```diff
@@ -107,6 +107,44 @@ export class KeybindingsEditorModel extends EditorModel {
 			return keybindingItems.map(keybindingItem => ({ id: KeybindingsEditorModel.getId(keybindingItem), keybindingItem, templateId: KEYBINDING_ENTRY_TEMPLATE_ID }));
 		}
 
+		if (this.isSourceFilterApplied(searchValue)) {
+			return this.filterBySource(keybindingItems, this.getSourceFilterValue(searchValue), completeMatch);
+		}
+		return this.filterByText(keybindingItems, searchValue, completeMatch);
+	}
+
+	private isSourceFilterApplied(searchValue: string): boolean {
+		return /^@source:/i.test(searchValue);
+	}
+
+	private getSourceFilterValue(searchValue: string): string {
+		return searchValue.split('@source:')[1].trim() || '';
+	}
+
+	private filterBySource(keybindingItems: IKeybindingItem[], searchValue: string, completeMatch: boolean): IKeybindingItemEntry[] {
+		const result: IKeybindingItemEntry[] = [];
```

Comment:
@shobhitchittora I think we are close good with the changes except for changing `filterBySource` method a bit. Instead of delegating the filtering to `KeybindingItemMatches` lets filter `keybindingItems` by source directly and map them filtered results to `IKeybindingItemEntry`.

Sorry for providing more comments :(

Reply:
@sandy081 Don't be sorry. I encourage comments / suggestions till it's the best. 😇

Reply:
Done.